pkg/streamingpromql/operator/instant_vector_function.go

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/promql/engine.go
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/promql/functions.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Prometheus Authors
+
+package operator
+
+import (
+	"context"
+	"math"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/promql/parser"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/pooling"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+// InstantVectorFunction performs a histogram_count over a range vector.
+type InstantVectorFunction struct {
```

Comment:
Maybe it would be clearer to call this `FunctionOverInstantVector`, and rename `RangeVectorFunction` to `FunctionOverRangeVector`? They both produce instant vectors, and I wonder if the name is confusing.